tests/test_init_wizard.py

```
@@ -117,3 +117,139 @@ def test_run_init_wizard_group_chat_without_topics(tmp_path: Path, monkeypatch)
+
+# ---- Gap 2: validation failure ----
+
+
+def test_run_init_wizard_validation_failure(tmp_path: Path, monkeypatch) -> None:
+    config_path = tmp_path / "config.yaml"
+
+    _patch_prompt_sequence(monkeypatch, ["token", 424242])
+    _patch_confirm_sequence(monkeypatch, [False, True, True, True])
+
+    def _boom(*_args, **_kwargs):
+        raise Exception("boom")
```

Ruff (0.15.5)

[warning] Create your own exception

(TRY002)